src/jupynium/events_control.py

```diff
@@ -522,6 +522,12 @@ def process_request_event(nvim_info: NvimInfo, driver, event):
         event[3].send(ret_obj)
         return True, None
 
+    elif event[1] == "kernel_connect_info":
+        driver.switch_to.window(nvim_info.window_handles[bufnr])
+        kernel_id = driver.execute_script("Jupyter.notebook.kernel.id")
```

Comment:
Do you also get this error?

```
Error executing Lua callback: [string "<nvim>"]:718: attempt to concatenate local 'kernel_id' (a userdata value)
stack traceback:
        [string "<nvim>"]:718: in function 'Jupynium_get_kernel_connect_shcmd'
        [string "<nvim>"]:726: in function <[string "<nvim>"]:723>
```

I think the kernel_id value is wrong. You should add `return`.

```python
        kernel_id = driver.execute_script("return Jupyter.notebook.kernel.id")
```

Reply:
Sorry I only test no-sync cases (connect to latest kernel) today since it's just implemented.

Reply:
That's okay. I think it works alright now. Thanks a lot for your work. The only things hasn't been done are

1. Test on Windows
2. Mention `pip install jupyter-console` dependency in README

```bash
# jupyter-console is optional and used for `:JupyniumKernelOpenInTerminal`
pip install notebook jupyter-console
```

I think you can add that command in the README Jupyter Notebook version dependency area. Then we're good to go

Reply:
I see that you have requested someone else to do the testing work for Windows. If they lack the time, since this plugin does not require any additional installation, I should be able to find an opportunity to complete the testing on Windows.